src/main/request.ts

```diff
@@ -78,21 +79,26 @@ export class Request {
     }
 
     async send(method: string, url: string, options: RequestOptions = {}): Promise<Response> {
-        const { retryAttempts, retryDelay } = this.config;
-        const attempts = retryAttempts < 1 ? 1 : retryAttempts + 1;
+        const totalAttempts = Math.max(this.config.retryAttempts + 1, 1);
         let attempted = 0;
         let lastError;
-        while (attempted < attempts) {
-            let shouldRetry = false;
+        while (attempted < totalAttempts) {
             attempted += 1;
+            let shouldRetry = false;
+            let retryDelay = this.config.retryDelay;
             try {
                 const res = await this.sendRaw(method, url, options);
                 if (!res.ok) {
-                    const invalidateAuth = this.matchStatusRange(res.status, this.config.statusCodeToInvalidateAuth);
+                    const invalidateAuth = this.config.authInvalidateStatusCodes.includes(res.status);
                     if (invalidateAuth) {
+                        // Note: we only retry once, if auth was authenticated
```

Comment:
This got me thinking, if we try only once, the logic with `clientSecret` will never run. so I checked the code and it's already unreachable at some cases, (we don't invalidate refreshToken in oauth2 so thus the next fallback logic with clientSecret will never be executed when `refreshToken` exists and is stale.

Reply:
Yep, this crossed my mind, but I didn't follow the thread. Do you think we should drop `resfreshToken` in `oauth2.invalidate()` then?

Reply:
This complicates the logic, but we can't invalidate `refreshToken` after first attempt, otherwise frontend-app which don't have clientSecret will never be able to get fresh accessToken 🤔

Reply:
Maybe we can revisit it later and go ahead as is, it's kind of separated that we need to address I believe.